test/test_write.c

```diff
@@ -602,7 +602,7 @@ int main(int argc, char **argv)
         /*hsize_t hchunk_dims[] = {19, 34};*/
         hsize_t hchunk_dims[] = {38, 128};
         hsize_t hchunk_off[] = {0, 0};
-        cfp_array2d* origarr;
+        cfp_array2d origarr;
```

Comment:
I haven't looked at this code in months but did I blunder this data type originally making it a pointer?

Reply:
It works as a pointer in zfp 0.5.5 but not in develop

Reply:
Are you saying ZFP changed its API for this datatype from having to be used as a pointer type in 0.5.5 to non-pointer type in their `develop` branch? Sorry if I am confused 😢

Reply:
I'm confused too. I'm not sure what changed in cfp. 

I found the reason why my tests failed locally, goes back to issue:
https://github.com/LLNL/H5Z-ZFP/issues/61

Issue : memory errors with H5Z-ZFP on 1.12.0 with free() but not H5MM_xfree()

Reply:
Looking at zfp code, it does look like develop switched the datatype, for example:
develop: void (*flush_cache)(const cfp_array2d self);
0.5.5: void (*flush_cache)(const cfp_array2d* self);

@lindstro  is that one of the changes for the next release?

Reply:
I fixed the test code to use the correct data type depending on the zfp version, but the issue is that zfp develop is versioned as 0.5.5 instead of being incremented.

Reply:
Yes, we changed the cfp types for the next release.  We're now supporting proxy pointers and iterators also in cfp, and it got a bit confusing to use `cfp_ptr*` to denote a pointer to an element, rather than just `cfp_ptr`, and `cfp_array**` to denote a pointer to an array, etc.  We also didn't want to dynamically allocate accessors like iterators, proxy references and pointers, and require them to be manually destructed.  So for consistency arrays are now `cfp_array` instead of `cfp_array*`, with the object pointer embedded in the struct.  As a result, the C API is now more closely aligned with the C++ API, e.g., `zfp::array2d` is equivalent to `cfp_array2d`.  See https://zfp.readthedocs.io/en/develop/cfp.html for further discussion.